# Author : Sunil Choudhary

In [1]:
# importing the required libraries.

import pandas as pd

In [2]:
df = pd.read_csv("TSF1.csv")

In [3]:
df.head()

Hours  Scores
0    2.5      21
1    5.1      47
2    3.2      27
3    8.5      75
4    3.5      30

The given data is about how much a student will score in exam/test if he spendthat much amount of time on his studies.

In [4]:
# Creating independent variable(x) and dependent variabe(y).

x = df.iloc[:,:-1].values
y = df.iloc[:, 1].values   

In [5]:
# Splitting the data into test data and train data.

from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x,y,random_state=0)

In [6]:
# Model fitting.

from sklearn.linear_model import LinearRegression  
regressor= LinearRegression()  
regressor.fit(x_train, y_train)  

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [7]:
#Prediction of Test and Training set result. 


regressor.predict(x_test)

array([16.84472176, 33.74557494, 75.50062397, 26.7864001 , 60.58810646,
       39.71058194, 20.8213931 ])

In [8]:
# Accurecy ?

print("accurecy of model is {} %".format(regressor.score(x_test,y_test)*100))

accurecy of model is 93.67661043365057 %


In [9]:
# Predicting the output for user defined input.

regressor.predict([[5.1]])

array([52.63476378])

The above line of code says that if student is studying for 5.1 Hours the he will score 52.63 marks.